shm/Cargo.hpp

```diff
@@ -18,5 +19,16 @@ class Cargo {
     std::string getName() const;
     size_t getAmount() const;
     size_t getBasePrice() const;
+    size_t getPrice() const {
+        return basePrice_;
+    }
 
+    std::unique_ptr<Cargo> split(size_t amountPart) {
+        if (!amountPart or amountPart > amount_) {
+            return {};
+        }
+
+        *this -= amountPart;
+        return std::make_unique<Cargo>(name_, amountPart, basePrice_);
+    }
```

Comment:
Is there a reason to put implementation of this method in header file?

Reply:
Moved implementation to cpp.